File: app/src/main/java/com/thanksmister/bitcoin/localtrader/ui/BaseFragment.java
/*
 * Copyright (c) 2018 ThanksMister LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed
 * under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.thanksmister.bitcoin.localtrader.ui;

import android.os.Bundle;

import com.thanksmister.bitcoin.localtrader.events.AlertDialogEvent;
import com.thanksmister.bitcoin.localtrader.events.ConfirmationDialogEvent;
import com.thanksmister.bitcoin.localtrader.events.ProgressDialogEvent;

import dagger.android.support.DaggerFragment;

/**
 * Base fragment which performs injection using the activity object graph of its parent.
 */
public abstract class BaseFragment extends DaggerFragment {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    protected void reportError(Throwable throwable) {
        if (isAdded() && getActivity() != null)
            ((BaseActivity) getActivity()).reportError(throwable);
    }

    protected void handleError(Throwable throwable, boolean retry) {
        if (isAdded() && getActivity() != null)
            ((BaseActivity) getActivity()).handleError(throwable, retry);
    }

    protected void handleError(Throwable throwable) {
        if (isAdded() && getActivity() != null)
            ((BaseActivity) getActivity()).handleError(throwable, false);
    }

    protected void toast(int messageId) {
        if (isAdded() && getActivity() != null)
            ((BaseActivity) getActivity()).toast(messageId);
    }

    protected void toast(String message) {
        if (isAdded() && getActivity() != null)
            ((BaseActivity) getActivity()).toast(message);
    }

    public void showAlertDialog(AlertDialogEvent event) {
        if (isAdded() && getActivity() != null)
            ((BaseActivity) getActivity()).showAlertDialog(event);
    }

    public void showConfirmationDialog(ConfirmationDialogEvent event) {
        if (isAdded() && getActivity() != null)
            ((BaseActivity) getActivity()).showConfirmationDialog(event);
    }

    public void showProgressDialog(ProgressDialogEvent event, boolean cancelable) {
        if (isAdded() && getActivity() != null)
            ((BaseActivity) getActivity()).showProgressDialog(event, cancelable);
    }

    public void showProgressDialog(ProgressDialogEvent event) {
        if (isAdded() && getActivity() != null)
            ((BaseActivity) getActivity()).showProgressDialog(event, false);
    }

    public void hideProgressDialog() {
        if (isAdded() && getActivity() != null)
            ((BaseActivity) getActivity()).hideProgressDialog();
    }
}

File: app/src/main/java/com/thanksmister/bitcoin/localtrader/ui/BaseFragment.kt
/*
 * Copyright (c) 2020 ThanksMister LLC
 *  http://www.thanksmister.com
 *
 *  Mozilla Public License 2.0
 *
 *  Permissions of this weak copyleft license are conditioned on making
 *  available source code of licensed files and modifications of those files
 *  under the same license (or in certain cases, one of the GNU licenses).
 *  Copyright and license notices must be preserved. Contributors provide
 *  an express grant of patent rights. However, a larger work using the
 *  licensed work may be distributed under different terms and without source
 *  code for files added in the larger work.
 */

package com.thanksmister.bitcoin.localtrader.ui

import android.os.Bundle
import com.thanksmister.bitcoin.localtrader.persistence.Preferences
import com.thanksmister.bitcoin.localtrader.utils.DialogUtils
import com.thanksmister.bitcoin.localtrader.utils.disposeProper

import dagger.android.support.DaggerFragment
import io.reactivex.disposables.CompositeDisposable
import javax.inject.Inject

/**
 * Base fragment which performs injection using the activity object graph of its parent.
 */
abstract class BaseFragment : DaggerFragment() {

    val disposable = CompositeDisposable()

    @Inject
    lateinit var preferences: Preferences

    @Inject
    lateinit var dialogUtils: DialogUtils

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        lifecycle.addObserver(dialogUtils)
    }

    override fun onDestroy() {
        super.onDestroy()
        disposable.disposeProper()
    }
}